base_layer/core/src/chain_storage/lmdb_db/lmdb.rs

```diff
@@ -75,6 +75,18 @@ where
         .map_err(|e| ChainStorageError::AccessError(e.to_string()))
 }
 
+pub fn lmdb_replace<K, V>(txn: &WriteTransaction, db: &Database, key: &K, val: &V) -> Result<(), ChainStorageError>
```

Comment:
Wouldn't it be simpler to have a `replace: bool` argument in `lmdb_insert` and

`flag = if replace { put::Flags::empty() } else { put::NOOVERWRITE };`

Reply:
Yes that will clean the code, I will implement it in a follow up PR.